### DISCUSSION TITLE
v0.3.0b1

### DISCUSSION THREAD

**Discussion by dynobo** [dynobo/normcap#187] [Announcements]
See [CHANGELOG](https://github.com/dynobo/normcap/blob/main/CHANGELOG.md) for details.

<hr /><em>This discussion was created from the release <a href='https://github.com/dynobo/normcap/releases/tag/unstable'>v0.3.0b1</a>.</em>

**Comment by dominikconrads** [comment]
![Screen Shot 2022-03-15 at 22 41 47](https://user-images.githubusercontent.com/36445834/158477514-b8fae822-d829-45fb-ac50-73edd9b4e53d.png)

**Reply by dynobo** [reply]
Hi @dominikconrads, thanks for bringing this up to discussion!

Unfortunatly, this is expected behavior on MacOS. You need to go through a lot of somewhat scary steps to allow NormCap to run. From the README:

> On MacOS, allow the unsigned application on first start: "System Preferences" → "Security & Privacy" → "General" → "Open anyway". You might also need to allow NormCap to take screenshots.

The background is, that Apple demands a 100$/year subscription (and a lot of personal data) to sign an app, and I am hesitant to foster such a practise.

If anyone has ideas to mitigate this issue, I would be super interested! Please also let me know if/how I can improve docu around this issue, so users are less confused. :-)

**Reply by dominikconrads** [reply]
Yes I know. But the suggested steps circumvents Apples Gatekeeper and massively weakens security measurements.
The behavior on windows is similar.
So that isn't a way I want to go.

Unfortunately its the only way for Apple/Microsoft to secure their platforms and if a developer want users to use their software, it seems its the price devs have to pay for.
There is not more "personal data" required as it needs for the establishment of a company.

**Reply by dynobo** [reply]
One alternative is to install [NormCap as Python](https://pypi.org/project/normcap/) package. However, it is more cumbersome and requires installing `tesseract` on your own, e.g., using `homebrew`, which also requires elevated rights. Or see if one of the [alternative applications](https://dynobo.github.io/normcap/#faqs-alternatives) offer a signed app for Mac.

Any concrete help is very welcome, especially from Mac-savvy people. Issue #135 would be the right place for that.

Sorry, that this is answer is not really satisfying... :confused:

**Reply by dominikconrads** [reply]
Thank you for the hints.
No problem.
In general app developers should rely more on security, then on writing code only.
